rpcs3/rpcs3qt/cheat_manager.cpp

```diff
@@ -0,0 +1,660 @@
+	QPushButton* btn_new_search           = new QPushButton(tr("New Search"));
+	btn_filter_results                    = new QPushButton(tr("Filter Results"));
+	btn_filter_results->setEnabled(false);
+	edt_cheat_search_value = new QLineEdit();
+	cbx_cheat_search_type  = new QComboBox();
+	cbx_cheat_search_type->insertItems(0, cheat_types);
+	cbx_cheat_search_type->setCurrentIndex(signed_32_cheat);
+	grp_add_cheat_sub_layout->addWidget(btn_new_search);
+	grp_add_cheat_sub_layout->addWidget(btn_filter_results);
+	grp_add_cheat_sub_layout->addWidget(edt_cheat_search_value);
+	grp_add_cheat_sub_layout->addWidget(cbx_cheat_search_type);
+	grp_add_cheat_layout->addLayout(grp_add_cheat_sub_layout);
+	lst_search = new QListWidget(this);
+	lst_search->setSelectionMode(QAbstractItemView::SelectionMode::SingleSelection);
+	lst_search->setSelectionBehavior(QAbstractItemView::SelectRows);
+	lst_search->setContextMenuPolicy(Qt::CustomContextMenu);
+	grp_add_cheat_layout->addWidget(lst_search);
+	grp_add_cheat->setLayout(grp_add_cheat_layout);
+	main_layout->addWidget(grp_add_cheat);
+
+	setLayout(main_layout);
+
+	// Edit/Manage UI
+	connect(tbl_cheats, &QTableWidget::itemClicked, [=](QTableWidgetItem* item) {
+		int row           = tbl_cheats->currentRow();
+		cheat_info* cheat = g_cheat.get(tbl_cheats->item(row, 0)->text().toStdString(), tbl_cheats->item(row, 3)->data(Qt::UserRole).toUInt());
+		if (cheat)
+		{
+			QString cur_value;
+			bool success;
+
+			u64 result_value;
+			switch (cheat->type)
+			{
+			case unsigned_8_cheat: result_value = g_cheat.get_value<u8>(cheat->offset, success); break;
+			case unsigned_16_cheat: result_value = g_cheat.get_value<u16>(cheat->offset, success); break;
+			case unsigned_32_cheat: result_value = g_cheat.get_value<u32>(cheat->offset, success); break;
+			case unsigned_64_cheat: result_value = g_cheat.get_value<u64>(cheat->offset, success); break;
+			case signed_8_cheat: result_value = g_cheat.get_value<s8>(cheat->offset, success); break;
+			case signed_16_cheat: result_value = g_cheat.get_value<s16>(cheat->offset, success); break;
+			case signed_32_cheat: result_value = g_cheat.get_value<s32>(cheat->offset, success); break;
+			case signed_64_cheat: result_value = g_cheat.get_value<s64>(cheat->offset, success); break;
+			default: log_cheat.fatal("Unsupported cheat type"); return;
+			}
+
+			if (success)
+			{
+				if (cheat->type >= signed_8_cheat && cheat->type <= signed_64_cheat)
+					cur_value = tr("%1").arg((s64)result_value);
+				else
+					cur_value = tr("%1").arg(result_value);
+
+				btn_apply->setEnabled(true);
+			}
+			else
+			{
+				cur_value = tr("Failed to get the value from memory");
+				btn_apply->setEnabled(false);
+			}
+
+			edt_value_final->setText(cur_value);
+		}
+		else
+		{
+			log_cheat.fatal("Failed to retrieve cheat selected from internal cheat_engine");
+		}
+	});
+
+	connect(tbl_cheats, &QTableWidget::cellChanged, [=](int row, int column) {
+		if (column != 1)
```

Comment:
also check for row maybe

Reply:
This is only called when a cell has been edited(after the edit) so I think it's ok like that?